sdk/keyvault/keyvault-certificates/src/core/challengeBasedAuthenticationPolicy.ts

```diff
@@ -82,102 +108,126 @@ export class ChallengeBasedAuthenticationPolicy extends BaseRequestPolicy {
     super(nextPolicy, options);
   }
 
-  private parseWWWAuthenticate(www_authenticate: string): {
-    authorization: string,
-    resource: string
-  } {
-    const returnValue = {
-      authorization: "",
-      resource: ""
-    };
-    // Parses an authentication message like:
-    // ```
-    // Bearer authorization="some_authorization", resource="https://some.url"
-    // ```
-    let spaceSep = www_authenticate.split(" ");
-
-    // Split the KV comma-separated list
-    for (const spaceItem of spaceSep) {
-      const commaSep = spaceItem.split(",");
-      for (const commaItem of commaSep) {
-        // Split the key/value pairs
-        const kv = commaItem.split("=");
-        const key = kv[0].trim();
-        const removeQuotes = (x: string): string => x.trim().replace(/['"]+/g, '');
-        if (key == "authorization" || key == "authorization_uri") {
-          returnValue.authorization = removeQuotes(kv[1]);
-        } else if (key == "resource" || key == "scope") {
-          returnValue.resource = removeQuotes(kv[1]);
-        }
-      }
+  /**
+   * Gets or updates the token from the token cache into the headers of the received web resource.
+   */
+  private async loadToken(webResource: WebResource): Promise<void> {
+    let accessToken = this.tokenCache.getCachedToken();
+
+    // If there's no cached token in the cache, we try to get a new one.
+    if (accessToken === undefined) {
+      const receivedToken = await this.credential.getToken(this.challengeCache.challenge!.scope);
+      accessToken = receivedToken || undefined;
+      this.tokenCache.setCachedToken(accessToken);
+    }
+
+    if (accessToken) {
+      webResource.headers.set(
+        Constants.HeaderConstants.AUTHORIZATION,
+        `Bearer ${accessToken.token}`
+      );
     }
-    return returnValue;
   }
 
   /**
-   * Applies the Bearer token to the request through the Authorization header.
-   * @param webResource
+   * Parses an WWW-Authenticate response.
+   * This transforms a string value like:
+   * `Bearer authorization="some_authorization", resource="https://some.url"`
+   * into an object like:
+   * `{ authorization: "some_authorization", resource: "https://some.url" }`
+   * @param wwwAuthenticate string value in the WWW-Authenticate header
+   */
+  public parseWWWAuthenticate(wwwAuthenticate: string): ParsedWWWAuthenticate {
```

Comment:
why is this public?

Reply:
if the answer is "for testing", why not split it out into a separate method that is exported?

Reply:
On it!